ml-conversational-analytic-tool/run.py

```diff
@@ -2,14 +2,30 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import argparse
+import tarfile
+import os
+
 from sklearn.model_selection import train_test_split
 
 from baseCNN import BaseCNN
 from baseLSTM import BaseLSTM
 from preProcessedDataset import PreProcessedDataset
 
-
-def run(annotated_filename, dataset_filename, outcome, encoding_type, model_type, padding):
+model_direcrtory = 'models'
+model_version = '0001'
+
+def save_model(model, name):
+    if not os.path.exists(model_direcrtory):
+        os.makedirs(model_direcrtory)
+    os.chdir(model_direcrtory)
+    model.saveModel(name=name, version=model_version)
+    os.chdir(name)
+    tar = tarfile.open(name + "-" + model_version + ".tar.gz", "w:gz")
```

Comment:
I don't think it's necessary to compress to tar.gz file.
By doing this, actually makes it harder to load the model. WDYT?

Reply:
I added a compression to simplify the model loading. In our case the model is not a single file, but a whole directory with a specific structure. When I tested this with kserve, I found that the predictor service expects exactly that directory layout with these files. I used my web server to pass the model to the predictor service:
`storageUri: "http://zico.biz/test/cat-model-0001.tar.gz"`
and the only way to make it work is to put a single archive there. The service downloads the archive, decompresses it and loads the files.
I agree that this is kserve specific, but I do not have other ml serving framework to test with. I guess that this is something specific to tensorflow models, and other serving framework should expect the same files.

Reply:
I was thinking not for deployment but for when you want to load the model back into your code. For example, when you want to run `keras.models.load_model("model")`

But, looks like you do keep the saved model intact so that it can be reloaded if the user chooses to. So, generating a tar file is just an additional functionality. 

Thanks for testing with the kserve, I think this is fine.